pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -227,7 +227,8 @@ def on_run_end(self) -> List[List[STEP_OUTPUT]]:
         self.trainer.call_hook("on_epoch_end")
         self.trainer.logger_connector.on_epoch_end()
 
-        self.update_lr_schedulers("epoch", update_plateau_schedulers=True)
+        if self._num_training_batches_reached(self.is_last_batch):
```

Comment:
`is_last_batch` wouldn't be necessary here, right? since it's `on_run_end`.

```suggestion
        if self._num_training_batches_reached():
```

I need to try and replace the saved attribute in favor of progress tracking